### PR TITLE
storage: add new ImportEpoch DeleteRangePredicate 

### PR DESCRIPTION
**Pull request by stevendanna**
When set, MVCCDeleteRangePredicate will delete any values with
ImportEpoch's in their MVCCValueHeader that is >= the given
ImportEpoch predicate.

Release note: none
Epic: None

### CODE REVIEW

**Comment by stevendanna** [comment]
All but the last commit is https://github.com/cockroachdb/cockroach/pull/85138

**Review by msbutler — COMMENTED** [review]
Glad we'll get this in before branch cut!

File: pkg/storage/mvcc.go [hunk 1]
```diff
@@ -3786,6 +3787,30 @@ func MVCCPredicateDeleteRange(
...
+		}
+	} else {
+		stopRunBasedOnPredicate = func(k MVCCKey, _ *MVCCIncrementalIterator) (bool, error) {
+			return k.Timestamp.LessEq(predicates.StartTime), nil
+		}
+	}
+
 	// continueRun returns three bools: the first is true if the current run
 	// should continue; the second is true if the latest key is a point tombstone;
 	// the third is true if the latest key is a range tombstone. If a non-nil
```
[the remark this answers is not in the document]
Comment (stevendanna) [reply pkg/storage/mvcc.go:3816]: I think this comment is still correct?

File: pkg/kv/kvpb/api.proto [hunk 2]
```diff
@@ -498,6 +498,10 @@ message DeleteRangeRequest {
 //
 // To pass DeleteRangePredicates, the client must also pass UseRangeTombstone.
 message DeleteRangePredicates {
+  // ImportEpoch specifies that all keys with a non-zero
+  // MVCCValueHeader.ImportEpoch >= ImportEmpoch shhould be deleted.
+  uint32 import_epoch = 1;
```
[the remark this answers is not in the document]
Comment (stevendanna) [reply pkg/kv/kvpb/api.proto:503]: Done.

File: pkg/storage/mvcc.go [hunk 3]
```diff
@@ -3786,6 +3787,30 @@ func MVCCPredicateDeleteRange(
...
+			if err != nil {
+				return true, err
+			}
+			v, err := DecodeMVCCValue(rawV)
+			if err != nil {
+				return true, err
+			}
+			return v.ImportEpoch < predicates.ImportEpoch, nil
+		}
+	} else {
```
[the remark this answers is not in the document]
Comment (stevendanna) [reply pkg/storage/mvcc.go:3807]: Done.

File: pkg/storage/mvcc.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/storage/mvcc.go:3815]: oof yes. misread endtime for starttime.

**Review by msbutler — APPROVED** [review]
LGTM!

**Comment by stevendanna** [comment]
bors r=msbutler

**Pull request merged**